diff --git a/src/executors/rollup/lib/update-package-json.js b/src/executors/rollup/lib/update-package-json.js
index 48283ecf5c190a40e0659ad448bc541163805574..e710c225a164d1ab970303eec7bcf2cd4101a0a4 100644
--- a/src/executors/rollup/lib/update-package-json.js
+++ b/src/executors/rollup/lib/update-package-json.js
@@ -17,14 +17,14 @@ function updatePackageJson(options, context, target, dependencies, packageJson)
     if (hasEsmFormat) {
         // `module` field is used by bundlers like rollup and webpack to detect ESM.
         // May not be required in the future if type is already "module".
-        packageJson.module = './index.js';
+        packageJson.module = './index.esm.js';
         exports['.']['import'] = './index.js';
         if (!hasCjsFormat) {
             packageJson.main = './index.js';
         }
     }
     if (hasCjsFormat) {
-        packageJson.main = './index.cjs';
+        packageJson.main = './index.cjs.js';
         exports['.']['require'] = './index.cjs';
     }
     if (!options.skipTypeField) {
diff --git a/src/executors/rollup/rollup.impl.js b/src/executors/rollup/rollup.impl.js
index 2ca97d37c2fa1fa54d3372e8200054cdf343ac08..6849b7a14ee91f6374385d22de9a5dda758354a7 100644
--- a/src/executors/rollup/rollup.impl.js
+++ b/src/executors/rollup/rollup.impl.js
@@ -196,8 +196,8 @@ function createRollupOptions(options, dependencies, context, packageJson, source
                 format,
                 dir: `${options.outputPath}`,
                 name: (0, devkit_1.names)(context.projectName).className,
-                entryFileNames: `[name].${format === 'esm' ? 'js' : 'cjs'}`,
-                chunkFileNames: `[name].${format === 'esm' ? 'js' : 'cjs'}`,
+                entryFileNames: `[name].${format === 'esm' ? 'esm' : 'cjs'}.js`,
+                chunkFileNames: `[name].${format === 'esm' ? 'esm' : 'cjs'}.js`,
             },
             external: (id) => externalPackages.some((name) => id === name || id.startsWith(`${name}/`)) || npmDeps.some((name) => id === name || id.startsWith(`${name}/`)),
             plugins,
