diff --git a/src/executors/rollup/rollup.impl.js b/src/executors/rollup/rollup.impl.js
index baca3a675716434819a482ace464c4afa9e51b5f..7861137cb48e7dbf09709dfbd09517ed898a7959 100644
--- a/src/executors/rollup/rollup.impl.js
+++ b/src/executors/rollup/rollup.impl.js
@@ -209,7 +209,7 @@ function createCompilerOptions(options, dependencies) {
 function updatePackageJson(options, context, target, dependencies, packageJson) {
     const entryFileTmpl = `./index.<%= extension %>.js`;
     const typingsFile = (0, path_1.relative)(options.entryRoot, options.entryFile).replace(/\.[jt]sx?$/, '.d.ts');
-    packageJson.main = entryFileTmpl.replace('<%= extension %>', 'umd');
+    packageJson.main = entryFileTmpl.replace('<%= extension %>', 'cjs');
     packageJson.module = entryFileTmpl.replace('<%= extension %>', 'esm');
     packageJson.typings = `./${typingsFile}`;
     (0, devkit_1.writeJsonFile)(`${options.outputPath}/package.json`, packageJson);
