diff --git a/src/utils/check-dependencies.js b/src/utils/check-dependencies.js
index 86c0489478e532dd9df89b553beeb72dc168627b..1cef2df06c291c435e7021c29575371973573931 100644
--- a/src/utils/check-dependencies.js
+++ b/src/utils/check-dependencies.js
@@ -3,7 +3,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.checkDependencies = void 0;
 const buildable_libs_utils_1 = require("@nrwl/workspace/src/utilities/buildable-libs-utils");
 function checkDependencies(context, tsConfigPath) {
-    const { target, dependencies } = (0, buildable_libs_utils_1.calculateProjectDependencies)(context.projectGraph, context.root, context.projectName, context.targetName, context.configurationName);
+    const { target, dependencies } = (0, buildable_libs_utils_1.calculateProjectDependencies)(context.projectGraph, context.root, context.projectName, context.targetName, context.configurationName, true);
     const projectRoot = target.data.root;
     if (dependencies.length > 0) {
         return {
