diff --git a/lib/packageManager/packagePublish.js b/lib/packageManager/packagePublish.js
index 6a211b2b472c8d16f776f79a89d5914cabd390b7..f0006dd42e1f8aae581e61c6c5b6dfe495500ed1 100644
--- a/lib/packageManager/packagePublish.js
+++ b/lib/packageManager/packagePublish.js
@@ -9,6 +9,10 @@ const npm_1 = require("./npm");
 function packagePublish(packageInfo, registry, token, access, authType, timeout) {
     const packageOptions = packageInfo.combinedOptions;
     const packagePath = path_1.default.dirname(packageInfo.packageJsonPath);
+
+    const packageDir = require('path').relative(packageInfo.combinedOptions.path, packagePath);
+    const artifactsPath = require('path').resolve(packageInfo.combinedOptions.path, 'dist', packageDir);
+
     const args = [
         'publish',
         '--registry',
@@ -23,8 +27,9 @@ function packagePublish(packageInfo, registry, token, access, authType, timeout)
         args.push('--access');
         args.push(access);
     }
+    console.log(`publish cwd: ${artifactsPath}`);
     console.log(`publish command: ${args.join(' ')}`);
-    return npm_1.npmAsync(args, { cwd: packagePath, timeout, all: true });
+    return npm_1.npmAsync(args, { cwd: artifactsPath, timeout, all: true });
 }
 exports.packagePublish = packagePublish;
 //# sourceMappingURL=packagePublish.js.map
