diff --git a/lib/Mermaid.mjs b/lib/Mermaid.mjs
index 5140cd5e556ede3debb33ae2202f04480b288d2f..204a3f97d590c9a384ed1bdcfcf7b0dc1375c616 100644
--- a/lib/Mermaid.mjs
+++ b/lib/Mermaid.mjs
@@ -45,7 +45,7 @@ const Mermaid = ({ chart, config: configSrc }) => {
     if (typeof window === 'undefined') {
         return React.createElement("div", { className: "mermaid", "data-mermaid-src": chart }, chart);
     }
-    const config = useMemo(() => typeof configSrc === 'string' ? JSON.parse(configSrc) : configSrc, [configSrc]);
+    const config = useMemo(() => typeof configSrc === 'string' ? JSON.parse(configSrc) : configSrc || {}, [configSrc]);
     const html = document.querySelector('html');
     const [rerender, setRerender] = useState(false);
     const theme = useMemo(() => getTheme(html, config), [config, rerender]);
